app/code/community/Bolt/Boltpay/Model/BoltOrder.php

```diff
@@ -1055,4 +1060,41 @@ private function dispatchCartDataEvent($eventName, $quote, &$cartSubmissionData)
         );
         $cartSubmissionData = $cartDataWrapper->getCartData();
     }
+
+    /**
+     * It is Magento part of code which allow to work
+     * with applied Catalog Price Rule in the
+     * back-office due order create process.
+     * The original path is \Mage_Adminhtml_Sales_Order_CreateController::_processActionData()
+     * and \Mage_Adminhtml_Model_Sales_Order_Create::initRuleData()
+     *
+     * @param $storeId
+     * @param $websiteId
+     * @param $customerGroupId
+     */
+    protected function initRuleData($storeId, $websiteId, $customerGroupId)
+    {
+        Mage::register('rule_data', new Varien_Object(array(
+            'store_id' => $storeId,
+            'website_id' => $websiteId,
+            'customer_group_id' => $customerGroupId,
+        )));
+    }
```

Comment:
I would like to offer a suggestion here that has two options.  

We can either move this function to the natural class for it which is Bolt_Boltpay_Adminhtml_Sales_Order_CreateController
https://github.com/BoltApp/bolt-magento1/blob/master/app/code/community/Bolt/Boltpay/controllers/Adminhtml/Sales/Order/CreateController.php

It will still be callable from the `Bolt_Boltpay_Controller_Traits_OrderControllerTrait`

However, i believe the best option is to get rid of this method in entirety and instead of making the conditional call to this custom made `initRuleData` here
https://github.com/BoltApp/bolt-magento1/blob/master/app/code/community/Bolt/Boltpay/Controller/Traits/OrderControllerTrait.php#L44

we instead conditionally call the real `Mage_Adminhtml_Model_Sales_Order_Create::initRuleData()` with the code
```
        if ($checkoutType === Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_ADMIN) {
               $this->_getOrderCreateModel()->initRuleData();
        }
```
from the same spot.  Because this is where the code naturally fits in, you will be able to use $this and will be referencing `Mage_Adminhtml_Sales_Order_CreateController`

Reply:
@mccottry Thanks, i will move to the `OrderControllerTrait`